### DISCUSSION TITLE
Should I contribute?

### DISCUSSION THREAD

**Discussion by xbc5** [org-roam/org-roam-ui#299] [Q&A]
This looks nice, and a graph is extremely helpful, but there's a number of things that make it unusable:

- loading a complete graph of ~4k nodes (I don't need to load everything all at once);
- lack of keyboard controls (my trackpad has been broken for years, so I cannot zoom);
- lack of snapping (the graph often sits at odd places on the canvas);
- the sidebar should not overlay the graph because it obscures it;

I'd like to work on these issues, but I don't want to waste my time on features that will sit in the PR queue for two years.

My question is: if I contribute, are these features likely to get merged?

**Comment by tefkah** [comment]
Hi! 

Those features would be very welcome! I'm in quite low maintenance mode atm, but at least 2, and 4 would be relatively easy to add and approve. 1 would require some more checking to see if everything still works, but seems like a good thing to add.

I'm not exactly sure what you mean by 3, but I would be open to it!

For 4: I agree, but I chose this because (afaik) it's not really possible to dynamically resize the canvas, it looks real awkward. I think it would still be good even if its awkward tho, so I would probably approve such a PR.

In general, glad to hear you are interested in contributing! I'm kinda busy, so if you do open a PR and I don't get to it within a week feel free to @ me!

**Reply by xbc5** [reply]
> For 4 .. it's not really possible to dynamically resize the canvas, it looks real awkward

The side-bar seems to be an important element of the UI, why not just make it permanently open, and make the canvas 66% the width of the screen (permanently)? No resizing then.

The alternative is to offset the h-centre a little when the side bar is open so that it doesn't "feel" like an overlay; and if we do 1. (local graph) we can reset the zoom level via a fit-to-canvas feature automatically, or at the users discretion via a quick and convenient keymap.

> 1 would require some more checking to see if everything still works, but seems like a good thing to add.

IIRC there's an existing feature that loads the local node, and the nearest N neighbours -- I was thinking of making this a more front-and-centre feature (toggled by a setting of course) such that accidentally loading the full graph (which can cause serious lag) requires purposeful intent, and is not easily done by accident.

> > lack of snapping (the graph often sits at odd places on the canvas);

> I'm not exactly sure what you mean by [the above], but I would be open to it! 

Like a keymap that snaps the view back to the local node; or when double clicking a node it snaps it to the centre more reliably.

Thanks for replying.
